transactions/src/test/java/org/jboss/hal/testsuite/test/transactions/TransactionCommitAndRollbackTestCase.java

```diff
@@ -0,0 +1,248 @@
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Vector;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Test class testing rollback and commit of prepared transactions in Web Console.
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@Category(Standalone.class)
+public class TransactionCommitAndRollbackTestCase {
+
+    private static final Logger logger = LoggerFactory.getLogger(TransactionCommitAndRollbackTestCase.class);
+
+    @Drone
+    private WebDriver browser;
+
+    @Page
+    private MessagingPreparedTransactionsPage page;
+
+    private static final OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient();
+    private static final Operations operations = new Operations(client);
+
+    private static final Address TRANSACTIONS_LOGGER_ADDRESS = Address.subsystem("logging").and("logger",
+            "org.apache.activemq.artemis.core.transaction.impl.TransactionImpl");
+
+    private static Path serverLog;
+
+    @BeforeClass
+    public static void beforeClass() throws IOException {
+        operations.add(TRANSACTIONS_LOGGER_ADDRESS, Values.of("level", "TRACE"));
+        serverLog = new PathOperations(client).getServerLogFile();
+    }
+
+    @AfterClass
+    public static void afterClass() throws IOException, OperationException {
+        try {
+            operations.removeIfExists(TRANSACTIONS_LOGGER_ADDRESS);
+        } finally {
+            client.close();
+        }
+    }
+
+    @Before
+    public void before() {
+        page.navigate();
+    }
+
+    private static String[] TRANSATIONS_XID = new String[]{
+            "AAAAAAAAAAAAAP__CigFZuOgfONXxVvAAAASdQAAAAEAAAAAAAAAAAAAAAAAAP__CigFZuOgfONXxVvAAAASdDEHAgIA",
+            "AAAAAAAAAAAAAP__CigFZuOgfONXxVvAAAASvwAAAAEAAAAAAAAAAAAAAAAAAP__CigFZuOgfONXxVvAAAASdDEHAgIA",
+            "AAAAAAAAAAAAAP__CigFZtXWO0pXxVwMAAAFpgAAAAEAAAAAAAAAAAAAAAAAAP__CigFZtXWO0pXxVwMAAAFSTEHAgIA",
+            "AAAAAAAAAAAAAP__CigFZtXWO0pXxVwMAAAFSgAAAAEAAAAAAAAAAAAAAAAAAP__CigFZtXWO0pXxVwMAAAFSTEHAgIA",
+            "AAAAAAAAAAAAAP__CigFZuOgfONXxVvAAAASbwAAAAEAAAAAAAAAAAAAAAAAAP__CigFZuOgfONXxVvAAAASbjEHAgIA",
+            "AAAAAAAAAAAAAP__CigFZtXWO0pXxVwMAAAFrgAAAAEAAAAAAAAAAAAAAAAAAP__CigFZtXWO0pXxVwMAAAFrTEHAgIA",
+            "AAAAAAAAAAAAAP__CigFZtXWO0pXxVwMAAAFsgAAAAEAAAAAAAAAAAAAAAAAAP__CigFZtXWO0pXxVwMAAAFrTEHAgIA"
+    };
```

Comment:
Please add comment explaining what the values mean, how they were counted. I guess they are based on the provided transaction journal. If it is so, add note about it.